### PR TITLE
FAD-6393 updated app to use google tag manager, adds page titles

### PR DESCRIPTION
**Pull request by jasonrhodes**

### CODE REVIEW

File: public/static/tenant-config/production.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ window.SP.productionConfig = {
     has_signup: true
   },
   gaTag: 'UA-111136819-2',
+  gtmId: 'GTM-TJ99DKC',
```
Comment (jasonrhodes) [inline public/static/tenant-config/production.js]: remove this here, set up a dev account for tst/stg

Comment (jasonrhodes) [reply public/static/tenant-config/production.js]: also note to self: don't forget the A-E/A-D changes needed for this ticket

File: src/components/googleTagManager/GoogleTagManager.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  trackPageview() {
+    const { location } = this.props;
+    const route = findRouteByPath(location.pathname);
+
+    window.dataLayer.push({
+      event: 'content-view',
+      'content-name': location.pathname + location.search, // duplicates angular 1.x ui-router "$location.url()" which is /path?plus=search
+      'content-title': route.title || location.pathname // duplicate angular 1.x $rootScope.stateData.title
```
Comment (jasonrhodes) [inline src/components/googleTagManager/GoogleTagManager.js]: We might want to remove this `||` fallback, not sure

File: src/config/routes.js [hunk 3]
```diff
@@ -157,289 +165,338 @@ const routes = [
...
+    layout: App,
+    title: 'Sending Domains'
   },
   {
     path: '/account/sending-domains/create',
     component: sendingDomains.CreatePage,
     condition: hasGrants('sending_domains/manage'),
-    layout: App
+    layout: App,
+    title: 'New Domain'
```
Comment (gwalls) [inline src/config/routes.js]: Could be a little ambiguous. Maybe add "Sending" for this and other sending domain pages. e.g."New Sending Domain"

File: src/config/routes.js [hunk 4]
```diff
@@ -157,289 +165,338 @@ const routes = [
...
-    condition: configFlag('featureFlags.has_signup')
+    condition: configFlag('featureFlags.has_signup'),
+    title: 'Choose Your Plan'
   },
   {
     path: '/onboarding/sending-domain',
     component: onboarding.SendingDomainPage,
-    condition: configFlag('featureFlags.has_signup')
+    condition: configFlag('featureFlags.has_signup'),
+    title: 'Create a Sending Domain'
```
Comment (gwalls) [inline src/config/routes.js]: Would it be helpful to preface these titles with "Onboarding"? Most of these onboarding page titles have equivalent pages inside of the app.

File: src/components/layout/Layout.js [hunk 5]
```diff
@@ -1,24 +1,21 @@
...
-    exact,
-    strict: false
-  }));
-
-  return React.createElement((route && route.layout) || Form, {}, children);
+  const route = findRouteByPath(location.pathname);
+  const LayoutComponent = route.layout || Form;
+  return (
+    <LayoutComponent>
+      {route.title && <Helmet><title>{route.title} | SparkPost</title></Helmet>}
```
Comment (gwalls) [inline src/components/layout/Layout.js]: missing test coverage for when route.title is defined

**Pull request merged**